ddtrace/sampler.py

```diff
@@ -254,20 +254,21 @@ def __init__(
         if rate_limit is None:
             rate_limit = int(os.getenv("DD_TRACE_RATE_LIMIT", default=self.DEFAULT_RATE_LIMIT))
 
-        # Ensure rules is a list
-        self.rules = []  # type: List[SamplingRule]
         if rules is None:
             env_sampling_rules = os.getenv("DD_TRACE_SAMPLING_RULES")
             if env_sampling_rules:
                 rules = self._parse_rules_from_env_variable(env_sampling_rules)
             else:
                 rules = []
+            self.rules = rules
+        else:
+            self.rules = []
```

Comment:
we mutate the rules below (when adding the default rule), change to use a new list and copy the rules rather than mutating the given rules list